backend/src/integrations/sync.ts

```diff
@@ -1654,6 +1666,399 @@ const syncSecretsGitLab = async ({
+					await addSecretVersion({
+						projectId: integration.appId,
+						secretId: key,
+						secretValue,
+					});
+				}
+			}
+		};
+
+		/**
+		 * Helper function to delete the list of secrets in gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {object} param.secrets
+		 * @returns
+		 */
+		const deleteSecrets = async ({
+			secrets,
+		}: {
+			secrets: Record<string, GCPSecretManager.ISecret>;
+		}) => {
+			for await (const key of Object.keys(secrets)) {
+				await deleteSecret({
+					projectId: integration.appId,
+					secretId: key,
+				});
+			}
+		};
+
+		/**
+		 * Encode string value to Buffer for gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.data
+		 * @returns {string}
+		 */
+		const encodeSecretValue = ({ data }: { data: string }): Buffer => {
+			const encodedSecretValue = Buffer.from(data);
+
+			return encodedSecretValue;
+		};
+
+		/**
+		 * Decode base64, uint8array to normal string for gcp-secret-manager
+		 *
+		 * @param {object} param
+		 * @param {string} param.data
+		 * @returns {string}
+		 */
+		const decodeSecretValue = ({
+			data,
+		}: {
+			data: string | Uint8Array;
+		}): string => {
+			let decodedSecretValue = "";
+
+			if (typeof data !== "string") {
+				decodedSecretValue = Buffer.from(data).toString("utf-8");
+
+				return decodedSecretValue;
+			}
+
+			decodedSecretValue = Buffer.from(data, "base64").toString("utf-8");
+
+			return decodedSecretValue;
+		};
+
+		/**
+		 * Helper function to get the raw secret key name from gcp-secret's path
+		 *
+		 * @param gcpSecretPath
+		 * @returns
+		 */
+		const getGCPSecretMangerKey = (gcpSecretPath: string) =>
+			gcpSecretPath.split("/").pop();
+
+		const gcpSecretManagerSecrets = await getSecretManagerSecrets({
+			projectId: integration.appId,
+		});
+
+		const keyToGCPSecret = gcpSecretManagerSecrets.reduce((prev, curr) => {
+			const gcpSecretMangerKey = getGCPSecretMangerKey(curr.name ?? "");
+
+			if (gcpSecretMangerKey) {
+				return { ...prev, [gcpSecretMangerKey]: curr };
+			}
+
+			return { ...prev };
+		}, {} as Record<string, GCPSecretManager.ISecret>);
+
+		const SECRETS_TO_ADD: Record<string, string> = {};
+		const SECRETS_TO_DELETE: Record<string, GCPSecretManager.ISecret> = {};
+		const SECRETS_TO_UPDATE: Record<string, string> = {};
+
+		for (const key of Object.keys(secrets)) {
+			if (keyToGCPSecret[key]) {
+				// adding the key-value to be updated in gcp
+				SECRETS_TO_UPDATE[key] = secrets[key];
+			} else {
+				// adding the key-value to be added in gcp
+				SECRETS_TO_ADD[key] = secrets[key];
+			}
+		}
+
+		for (const key of Object.keys(keyToGCPSecret)) {
+			if (!secrets[key]) {
+				// adding the key-value to be deleted in gcp
+				SECRETS_TO_DELETE[key] = keyToGCPSecret[key];
+			}
+		}
+
+		await Promise.allSettled([
+			addSecrets({ secrets: SECRETS_TO_ADD }),
+			updateSecrets({ secrets: SECRETS_TO_UPDATE }),
+			deleteSecrets({ secrets: SECRETS_TO_DELETE }),
```

Comment:
I think we should just be passing in an array of "secret keys" here instead of an object

Reply:
For `deleteSecrets` function ?